casanovo/utils.py

```diff
@@ -251,3 +254,28 @@ def log_sequencing_report(
         logger.info(
             "Median Peptide Length: %d", run_report["median_sequence_length"]
         )
+
+
+def check_dir(dir: pathlib.Path, file_patterns: Iterable[str]) -> None:
+    """
+    Check that no file names in dir match any of file_patterns
+
+    Parameters
+    ----------
+    dir : pathlib.Path
+        The directory to check for matching file names
+    file_patterns : Iterable[str]
+        UNIX style wildcard pattern to test file names against
+
+    Raises
+    ------
+    FileExistsError
+        If matching file name is found in dir
+    """
+    for pattern in file_patterns:
+        matches = list(dir.glob(pattern))
+        if len(matches) > 0:
+            raise FileExistsError(
+                f"File {matches[0].name} already exists in {dir} "
```

Comment:
**thought:** This might be a slightly misleading error message. The user could remove the file that's mentioned, but there could be multiple files matching the pattern, which are not explicitly mentioned. Rather than reporting all files, which might result in an unwieldy error message, maybe just report the pattern instead?

```
if next(dir.glob(pattern), None) is None:
    raise FileExistsError(...)
```

This also avoids having to consume the whole generator (which admittedly doesn't really matter that much, even if there are dozens to hundreds of checkpoints matching).

Reply:
Good call!